subsys/net/openthread/rpc/server/ot_rpc_diag_server.c

```diff
@@ -0,0 +1,165 @@
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ot_rpc_ids.h>
+
+#include <nrf_rpc_cbor.h>
+#include <string.h>
+
+#include <openthread/dataset.h>
+#include <openthread/instance.h>
+#include <openthread/link.h>
+#include <openthread/thread.h>
+#include <openthread/netdata.h>
+#include <zephyr/net/openthread.h>
+
+NRF_RPC_GROUP_DECLARE(ot_group);
+
+static size_t ret_size_str(const char *in)
+{
+	return strlen(in);
+}
+
+static size_t ret_size_ext_addr(const otExtAddress *in)
+{
+	return OT_EXT_ADDRESS_SIZE;
+}
+
+static size_t ret_size_ext_panid(const otExtendedPanId *in)
+{
+	return OT_EXT_PAN_ID_SIZE;
+}
+
+static size_t ret_size_mesh_local_prefix(const otMeshLocalPrefix *in)
+{
+	return OT_IP6_PREFIX_SIZE;
+}
+
+#define ret_size(X)                                                                               \
+	_Generic((X),                                                                             \
+		const char * : ret_size_str,                                                       \
+		const otExtAddress * : ret_size_ext_addr,                                          \
+		const otExtendedPanId * : ret_size_ext_panid,                                      \
+		const otMeshLocalPrefix * : ret_size_mesh_local_prefix)(X)
+
+static bool serialize_str(zcbor_state_t *state, const char *ser)
+{
+	size_t length = ret_size(ser);
+
+	return zcbor_bstr_encode_ptr(state, ser, length);
+}
+
+static bool serialize_extaddr(zcbor_state_t *state, const otExtAddress *ser)
+{
+	size_t length = ret_size(ser);
+
+	return zcbor_bstr_encode_ptr(state, ser->m8, length);
+}
+
+static bool serialize_ext_panid(zcbor_state_t *state, const otExtendedPanId *ser)
+{
+	size_t length = ret_size(ser);
+
+	return zcbor_bstr_encode_ptr(state, ser->m8, length);
+}
+
+static bool serialize_mesh_local_prefix(zcbor_state_t *state, const otMeshLocalPrefix *ser)
+{
+	size_t length = ret_size(ser);
+
+	return zcbor_bstr_encode_ptr(state, ser->m8, length);
+}
+
+#define serialize_bstr(Z, X)                                                                       \
+	_Generic((X),                                                                              \
+		const char * : serialize_str,                                                      \
+		const otExtAddress * : serialize_extaddr,                                          \
+		const otMeshLocalPrefix * : serialize_mesh_local_prefix,                           \
+		const otExtendedPanId  *  : serialize_ext_panid)(Z, X)
+
+#define NRF_RPC_CBOR_CMD_STRING_GETTER(func_name, func_to_call, call_id, type, ptr_call)           \
+	static void func_name(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,     \
+			      void *handler_data)                                                  \
+	{                                                                                          \
+		size_t length = 16;                                                                \
```

Comment:
Why is it 16?

Reply:
This is the reason: https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf_rpc/entropy_nrf53/cpuapp/src/entropy_ser.c#L19 so all in al no perticular one i can change it.

Reply:
Changed to @Damian-Nordic suggestion.